### PR TITLE
go timeout && subpatrition calculate incorrect

### PR DESCRIPTION
**Pull request by foronedream**
go timeout && subpatrition calculate incorrect
## Summary

go timeout && subpatrition calculate incorrect

## Solution Description

go timeout && subpatrition calculate incorrect

### CODE REVIEW

File: protocol/rpc_header.go [hunk 1]
```diff
@@ -395,7 +395,7 @@ func (h *ObRpcHeader) Decode(buffer *bytes.Buffer) {
 	h.sessionId = util.Uint64(buffer)
 	h.traceId0 = util.Uint64(buffer)
 	h.traceId1 = util.Uint64(buffer)
-	h.timeout = time.Duration(util.Uint64(buffer))
+	h.timeout = time.Duration(util.Uint64(buffer) * 1e6)
```
Comment (IHEII) [inline protocol/rpc_header.go:398]:
```suggestion
	h.timeout = time.Duration(util.Uint64(buffer)) * time.Millisecond
```

File: protocol/rpc_header_test.go [hunk 2]
```diff
@@ -86,7 +86,7 @@ func TestObRpcHeaderEncodeDecode(t *testing.T) {
 	assert.EqualValues(t, obRpcHeader.SessionId(), newObRpcHeader.SessionId())
 	assert.EqualValues(t, obRpcHeader.TraceId0(), newObRpcHeader.TraceId0())
 	assert.EqualValues(t, obRpcHeader.TraceId1(), newObRpcHeader.TraceId1())
-	assert.EqualValues(t, obRpcHeader.Timeout(), newObRpcHeader.Timeout())
+	assert.EqualValues(t, obRpcHeader.Timeout().Milliseconds(), newObRpcHeader.Timeout()/1e6)
```
Comment (IHEII) [inline protocol/rpc_header_test.go:89]: Why use milliseconds in rpcHeader and use /1e6 in newRpcHeader?

File: protocol/rpc_header.go (hunk 1 above)
Comment (foronedream) [reply protocol/rpc_header.go]: edit done

File: protocol/rpc_header_test.go (hunk 2 above)
Comment (foronedream) [reply protocol/rpc_header_test.go]: edit

**Comment by IHEII** [comment]
LGTM

**Pull request merged**